cloudify_agent/operations.py

```diff
@@ -557,3 +558,16 @@ def _create_package_url(url, ip):
     after = rest.split(':')[1]
     new = '{before}//{ip}:{after}'.format(before=before, ip=ip, after=after)
     return new
+
+
+def _get_network_ip(manager_ip):
+    network_ip_mapping = yaml.safe_load(manager_ip)
+    # Multi-network mode: a dict containing networks and ips was passed
+    if type(manager_ip) is dict:
+        agent = ctx.instance.runtime_properties['cloudify_agent']
+        network_name = agent['network']
+        new_network_ip = network_ip_mapping[network_name]
+        return new_network_ip
+    # Regular mode: the only ip of the new manager was passed
+    else:
```

Comment:
do `elif` with a `string` (or maybe `basestring`), and `else` that raises an error